upload-api/stores/agent/stream.js

```diff
@@ -82,6 +82,25 @@ export const write = async (connection, message) => {
  * @param {Stream} connection.stream
  */
 export const assert = async (message, { stream, store }) => {
+  /** @param {API.UnknownLink} task */
+  const findInvocation = async (task) => {
+    // is the invocation in the message?
+    for (const member of message.index) {
+      if (member.invocation?.task.toString() === task.toString()) {
+        return member.invocation.invocation
+      }
+    }
+    // else find in store
+    const result = await Store.getInvocation(store, task)
+    if ('error' in result) {
+      console.error(result.error)
+      throw new NoInvocationFoundForGivenReceiptError(
+        `missing invocation: ${task}`
+      )
+    }
+    return result.ok
+  }
```

Comment:
This change simply allows the invocation we're trying to find to be contained in the agent message. i.e. invocation and receipt in the same message. This wouldn't usually happen but in the blob protocol we create fake invocations to allow the old client to work with the new capabilities (e.g. old client expects a `web3.storage/blob/allocate` invocation but it has become `blob/allocate`).